### ISSUE TITLE
Use a better database

### ISSUE THREAD

**Issue by 500-internal-server-error** [binusgdc/VCPA#19]
Currently, the bot uses a plain array to store session data. This makes the sessions "database" ephemeral and prone to disruptions (e.g., if the bot crashses while in the middle of a session, that session data is lost, as if the session was never started).

One solution would be to make use of sqlite as a more permanent database, allowing sessions to be resumed should they be interrupted for whatever reason. While it has never happened before as of the writing of this issue, it is a good precaution to take.

**Comment by saniistamar** [comment]
sqlite might be too overkill for such a scenario. Since the current implementation is just an in-memory key-value store, why not swap it out for an embedded one like [RocksDB](https://rocksdb.org/)? [levelup](https://www.npmjs.com/package/levelup) is an NPM package for interfacing with KV stores including RocksDB.

**Comment by 500-internal-server-error** [comment]
An alternative database is fine, as long as it's persistent, i.e., session data is not lost should the bot crash / be shut down for whatever reason.

**Comment by saniistamar** [comment]
Right, then I'll ~~get started on this~~ find someone to assign this to.

**Comment by saniistamar** [comment]
Use a Redis db as a buffer for completed sessions.

**Comment by alfonsusac** [comment]
Gw saranin langsung pake Prisma + Supabase/Neon aja
